src/Sfa.Das.Sas.Indexer.ApplicationServices/Provider/Services/TokenService.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Sfa.Das.Sas.Indexer.ApplicationServices.Shared.Settings;
+
+namespace Sfa.Das.Sas.Indexer.ApplicationServices.Provider.Services
+{
+    public class TokenService : ITokenService
+    {
+        private readonly IAppServiceSettings _settings;
+          public TokenService(IAppServiceSettings settings) 
+        {
+            _settings = settings;
+        }
+
+        public string GetToken()
```

Comment:
depending on how chatty this is, might be worth caching the response from here as the token is valid for 5 minutes i think?

Reply:
Nah, the call is  once a day to a single endpoint.  It might be 2-4 times a day if something broken, but it's definitely not going to be every 5 minutes, and the database can handle it anyway.